### PR TITLE
API: Added FLSlot_SetArray, FLSlot_SetDict

### PR DESCRIPTION
**Pull request by snej**
Simple inlines to avoid having to cast to (FLValue)

### CODE REVIEW

File: API/fleece/Fleece.h [hunk 1]
```diff
@@ -368,6 +368,14 @@ extern "C" {
     void FLSlot_SetData(FLSlot FLNONNULL, FLSlice) FLAPI;    ///< Stores a data blob into a slot.
     void FLSlot_SetValue(FLSlot FLNONNULL, FLValue) FLAPI;   ///< Stores an FLValue into a slot.
 
+    static inline void FLSlot_SetArray(FLSlot NONNULL slot, FLArray array) {
```
Comment (borrrden) [inline API/fleece/Fleece.h:371]: I wonder if array should be non-null.  Would a null array be a shortcut for `SetNull` ?

**Review by borrrden — CHANGES_REQUESTED** [review]
Could we have the mutable variants too? :)

**Comment by borrrden** [comment]
Does #95 supersede this?

**Comment by snej** [comment]
> Could we have the mutable variants too? :)

`FLMutableArray` is just a non-const `FLArray`, so they're type compatible already.

**Comment by snej** [comment]
> Does #95 supersede this?

No, it just means more inlines for me to write...

**Review by borrrden — APPROVED** [review]

**Pull request merged**